docker-compose.yml

```diff
@@ -1,58 +1,103 @@
 version: "3.9"
 
 services:
-  scheduler:
-    image: ghcr.io/traewelling/traewelling:latest
-    container_name: traewelling-scheduler
+  app:
+    build: .
     restart: 'always'
-    depends_on:
-      - app
+    ports:
+      - "8081:80"
     networks:
       - internal
     env_file:
       - .env.docker
     environment:
-      CONTAINER_ROLE: scheduler
+      CONTAINER_ROLE: app
+    volumes:
+      - ./app:/var/www/html/app
+      - ./config:/var/www/html/config
+      - ./resources:/var/www/html/resources
+      - ./routes:/var/www/html/routes
+      - ./storage:/var/www/html/storage
+      - ./tests:/var/www/html/tests
+      - ./avatars:/var/www/html/public/uploads/avatars
 
-  queue:
-    image: ghcr.io/traewelling/traewelling:latest
-    container_name: traewelling-queue
+      ## Un-comment the following line if you need to deal with frontend code.
+      ## The built assets will be overwritten by whatever is in your host
+      ## system, so run a local `npm run dev` on your host system to compile
+      ## your changed assets. The app container works as the webserver, and the
+      ## scheduler and the queue handler don't need the frontend dependencies
+      ## at this time.
+      ## Please don't commit the un-commenting to keep the dependencies
+      ## on the host computer as small as possible.
+      #- ./public:/var/www/html/public
+
+  scheduler:
+    build: .
     restart: 'always'
-    depends_on:
-      - app
     networks:
       - internal
     env_file:
       - .env.docker
     environment:
-      CONTAINER_ROLE: queue
+      CONTAINER_ROLE: scheduler
+    volumes:
+      - ./app:/var/www/html/app
+      - ./config:/var/www/html/config
+      - ./resources:/var/www/html/resources
+      - ./routes:/var/www/html/routes
+      - ./storage:/var/www/html/storage
+      - ./tests:/var/www/html/tests
+      - ./avatars:/var/www/html/public/uploads/avatars
 
-  app:
-    image: ghcr.io/traewelling/traewelling:latest
-    container_name: traewelling-app
+  queue:
+    build: .
     restart: 'always'
-    ports:
-      - "80:80"
     networks:
       - internal
     env_file:
       - .env.docker
     environment:
-      - CONTAINER_ROLE=app
+      CONTAINER_ROLE: queue
+    volumes:
+      - ./app:/var/www/html/app
+      - ./config:/var/www/html/config
+      - ./resources:/var/www/html/resources
+      - ./routes:/var/www/html/routes
+      - ./storage:/var/www/html/storage
+      - ./tests:/var/www/html/tests
+      - ./avatars:/var/www/html/public/uploads/avatars
 
   database:
     image: mariadb:10
-    container_name: traewelling-db
     restart: 'always'
+    ports:
+      - "3306:3306"
     volumes:
-      - ./docker/database:/var/lib/mysql
+      - ./database/docker-container-state:/var/lib/mysql
     networks:
       - internal
     environment:
       - TZ=Europe/Berlin
     env_file:
       - .env.docker
 
+  db-rest:
+    # we are stuck on db-rest version 5,
```

Comment:
I'm using arm architecture, and the docker images for this architecture were only recently added to the db-rest registry. But due to the version pinning I can't use them. Can you explain the reason for pinning the version?

Reply:
It should be fine as long as we're using v5, the pinning was because I was certain that it worked with that version.

Reply:
Resolved with 1cd34720